R/install-dev.R

```diff
@@ -56,7 +56,7 @@ install_dev <- function(package, cran_url = getOption("repos")[["CRAN"]], ...) {
         domain = paste0(valid_domains, collapse = "|"),
         username = "[^/]+",
         repo = "[^/@#]+",
-        subdir = "[^/@$ ]+"
+        subdir = "[^/@$ ]*"
```

Comment:
why do we need this?

Reply:
We don't right now, but perhaps we want to add a terminal `$` to the regex?

``` r
library(rematch2)
pkg_urls <- "https://github.com/r-quantities/units/"

valid_domains <- c("github[.]com", "gitlab[.]com", "bitbucket[.]org")

re_match(
  pkg_urls,
  sprintf("^https?://(?<domain>%s)/(?<username>%s)/(?<repo>%s)(?:/(?<subdir>%s))?",
    domain = paste0(valid_domains, collapse = "|"),
    username = "[^/]+",
    repo = "[^/@#]+",
    subdir = "[^/@$ ]+"
  )
)
#> # A tibble: 1 × 6
#>   domain     username     repo  subdir .text                              .match
#>   <chr>      <chr>        <chr> <chr>  <chr>                              <chr> 
#> 1 github.com r-quantities units ""     https://github.com/r-quantities/u… https…

re_match(
  pkg_urls,
  sprintf("^https?://(?<domain>%s)/(?<username>%s)/(?<repo>%s)(?:/(?<subdir>%s))?",
    domain = paste0(valid_domains, collapse = "|"),
    username = "[^/]+",
    repo = "[^/@#]+",
    subdir = "[^/@$ ]*"
  )
)
#> # A tibble: 1 × 6
#>   domain     username     repo  subdir .text                              .match
#>   <chr>      <chr>        <chr> <chr>  <chr>                              <chr> 
#> 1 github.com r-quantities units ""     https://github.com/r-quantities/u… https…

re_match(
  pkg_urls,
  sprintf("^https?://(?<domain>%s)/(?<username>%s)/(?<repo>%s)(?:/(?<subdir>%s))?$",
    domain = paste0(valid_domains, collapse = "|"),
    username = "[^/]+",
    repo = "[^/@#]+",
    subdir = "[^/@$ ]+"
  )
)
#> # A tibble: 1 × 6
#>   domain username repo  subdir .text                                  .match
#>   <chr>  <chr>    <chr> <chr>  <chr>                                  <chr> 
#> 1 <NA>   <NA>     <NA>  <NA>   https://github.com/r-quantities/units/ <NA>
```

<sup>Created on 2022-01-25 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
If we don't need it, then I would not change it. remotes is particularly well tested, so it is better to avoid changes if possible.